apps/vscode/command_client/command_client.py

```diff
@@ -368,3 +377,63 @@ def trigger_command_server_command_execution():
 class LinuxUserActions:
     def trigger_command_server_command_execution():
         actions.key("ctrl-shift-alt-p")
+
+
+@global_ctx.action_class("user")
+class GlobalUserActions:
+    def emit_pre_phrase_signal():
+        pass
+
+
+@ctx.action_class("user")
+class UserActions:
+    def emit_pre_phrase_signal():
+        get_signal_path("prePhrase").touch()
+
+
+class MissingCommunicationDir(Exception):
+    pass
+
+
+def get_signal_path(name: str) -> Path:
+    """
+    Get the path to a signal in the signal subdirectory.
+
+    Args:
+        name (str): The name of the signal
+
+    Returns:
+        Path: The signal path
+    """
+    communication_dir_path = get_communication_dir_path()
+
+    if not communication_dir_path.exists():
+        raise MissingCommunicationDir()
+
+    signal_dir = communication_dir_path / "signals"
+    signal_dir.mkdir(parents=True, exist_ok=True)
+
+    return signal_dir / name
+
+
+did_emit_pre_phrase_signal = False
```

Comment:
Personally I'm not too found of global/module level variables in the middle of the file. Personally I would move this to the top.

Reply:
Was trying to keep it close to where it's used but sure fair enough. Happy to have global variables at the top be our convention

Reply:
In this case I absolutely think so since it's used by the action did_emit_pre_phrase_signal() above it.